lib/test/t_CubaIntegration_std.cxx

```diff
@@ -0,0 +1,83 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief The test file of class CubaIntegration
+ *
+ *  Copyright 2005-2024 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/OT.hxx"
+#include "openturns/OTtestcode.hxx"
+#include "openturns/CubaIntegration.hxx"
+#include <cmath>
+
+using namespace OT;
+using namespace OT::Test;
+
+int main(int, char *[])
+{
+  TESTPREAMBLE;
+  OStream fullprint(std::cout);
+  setRandomGenerator();
+
+  try
+  {
+    // Testing four algorithms on a function from R^3 to R.
+    {
+      const SymbolicFunction f({"x", "y", "z"}, {"sin(x) * cos(y) * exp(z)"});
+      const Point lbIntegration({0., 0., 0.});
+      const Point ubIntegration({1., 1., 1.});
+      const String routines[] = {"cuhre", "divonne", "suave", "vegas"};
+      const Scalar valueRef = -sin(1.) * (cos(1.) - 1.) * (M_E - 1.);
+      for (UnsignedInteger i = 0 ; i < 4 ; i++)
+      {
+          CubaIntegration algo(routines[i]);
+          const Point value(algo.integrate(f, Interval(lbIntegration, ubIntegration)));
+          assert_almost_equal(value[0], valueRef, 1.E-3, 1.E-3);
+      }
+    }
+
+    // Testing three algorithms on a function from R^3 to R^2.
+    {
+      const SymbolicFunction f({"x", "y", "z"}, {"exp(-(x * x + y * y + z *z))", "(x / 2 * x / 2 + y / 3 * y / 3 + z / 4 * z / 4 - 1.125)"});
```

Comment:
This line is very long: please split it onto several lines if necessary to improve reading.